packages/nimble-components/src/table/index.ts

```diff
@@ -162,6 +217,61 @@ export class Table<
 
         this.updateTableOptions({ columns: generatedColumns });
     }
+
+    private updateVirtualizer(): void {
+        const options = this.createVirtualizerOptions();
+        if (this.virtualizer) {
+            this.virtualizer.setOptions(options);
+        } else {
+            this.virtualizer = new Virtualizer(options);
+        }
+        this.virtualizer._willUpdate();
+        this.handleVirtualizerChange();
+    }
+
+    private createVirtualizerOptions(): VirtualizerOptions<
+    HTMLElement,
+    HTMLElement
+    > {
+        const rowHeight = parseFloat(controlHeight.getValueFor(this));
+        return {
+            count: this.tableData.length,
+            getScrollElement: () => {
+                return this.viewport;
+            },
+            estimateSize: (_: number) => rowHeight,
+            enableSmoothScroll: true,
+            overscan: 3,
+            scrollToFn: elementScroll,
+            observeElementOffset,
+            observeElementRect,
+            onChange: () => this.handleVirtualizerChange()
+        } as VirtualizerOptions<HTMLElement, HTMLElement>;
+    }
+
+    private handleVirtualizerChange(): void {
+        const virtualizer = this.virtualizer!;
+        this.visibleItems = virtualizer.getVirtualItems();
+        this.allRowsHeight = virtualizer.getTotalSize();
+        // We're using a separate div ('table-scroll') to represent the full height of all rows, and
+        // the row container's height is only big enough to hold the virtualized rows. So we don't
+        // use the TanStackVirtual-provided 'start' offset (which is in terms of the full height)
+        // to translate every individual row, we just translate the row container.
+        let rowContainerTransform = '';
+        if (this.visibleItems.length > 0) {
+            const firstItem = this.visibleItems[0]!;
+            const lastItem = this.visibleItems[this.visibleItems.length - 1]!;
+            if (lastItem.end < this.allRowsHeight) {
+                const offsetY = firstItem.start - this.viewport.scrollTop;
+                rowContainerTransform = `translateY(${offsetY}px)`;
```

Comment:
I'm pretty sure that `translateY` is going to cause problems when we have content that opens from within a row (such as an action menu or a select).

Reply:
Select popups and menu button popups still look like they appear in the right spot, when we're scrolled in-between rows (i.e. a translate transform is active):
![TableTranslateAndSelectPopup](https://user-images.githubusercontent.com/20709258/213285544-9aaeb3fe-3186-45e8-a818-535b4768b450.png)
![TableTranslateAndMenuButton](https://user-images.githubusercontent.com/20709258/213285561-1b9b0ccf-b6a2-4c53-8e9a-33aa6d95a9b6.png)

Side note, I had to disable the `overflow: hidden` style on the `.cell-content-container` for the `nimble-select` dropdown popup to appear correctly, but other than that it looked fine / position looked correct.
There'll be some follow-on work to make sure that those popups are closed when you scroll (similar to @m-akinc 's comment on Teams about the text in cells incorrectly staying selected after scrolling) though.

Reply:
Great, thanks for exploring this. It sounds like we're setting ourselves up well for adding addition controls in the table, and we'll work through any additional issues (like dealing with scrolling) as they come up.